### ISSUE TITLE
Edit Library can't save

### ISSUE THREAD

**Issue by Mars160** [jellyfin/jellyfin-web#4945]
**Describe The Bug**

I edited a library and click the submit button, and there is a blue circle means loading.
And the circle is always here not disappear.
The chrome console said:
![image](https://github.com/jellyfin/jellyfin-web/assets/74127225/7477ada7-783e-49c8-8c8b-678889f3ae0d)

**Steps To Reproduce**

1. Go to 'Console'
2. Click on 'Library'
3. Choose a library and click manage the library
4. Cancel some metadata downloader
5. submit
6. See error

**Expected Behavior**

It should save it and take effect.

**Logs**

![image](https://github.com/jellyfin/jellyfin-web/assets/74127225/ed281edf-a330-41a1-b1de-3109b335ed83)

**Screenshots**

![image](https://github.com/jellyfin/jellyfin-web/assets/74127225/29069130-4cd9-4001-81ad-da9b52c18487)

**System (please complete the following information):**
 - Platform: Windows
 - Browser: Chrome
 - Jellyfin Version: 10.8.12.0

**Additional Context**

**Comment by dmitrylyzo** [comment]
For those who want to reproduce the "null name".
I was able to get such broken state on an unstable server with Skia update (the one that crashes).
When you create a library, the server creates a folder for it (in `/data/root/default/`) and crashes. We see the new library, but it has no id (="null name").

**Comment by joshuaboniface** [comment]
Ran into this as well using the latest unstables (10.9.0, build version `2024041505`) after upgrading my instance from 10.8.13 (and previous versions).

The issue was that the `Path` entries in the `TypedBaseItems` table of the `library.db` for each library did not match the actual filesystem paths.

My data is in `/srv/jellyfin`, with a symlink from `/var/lib/jellyfin` to `/srv/jellyfin`, which seemed to be enough in 10.8.z. But this resulted in TypedBaseItems entries that looked like this:

```
sqlite> select Path from TypedBaseItems where Path like '%/jellyfin/root/default/%';
/var/lib/jellyfin/root/default/Collections
/var/lib/jellyfin/root/default/Movies
/var/lib/jellyfin/root/default/Music
/var/lib/jellyfin/root/default/Shows
/var/lib/jellyfin/root/default/YouTube
```

I suspect due to having the data directory path configured wrong when initially setting up the libraries. 

Manually updating these DB entries to the correct paths corrected the issue. I no longer get the "null name" error and library settings save correctly.

**Comment by joshuaboniface** [comment]
At this point in the release cycle, I think this is probably an esoteric enough issue to leave as-is for now. We should definitely improve the *error handling* portion, but the actual bug we'll have to evaluate after release to see if it's majorly affecting people and fix it if so.

**Comment by gnattu** [comment]
> Ran into this as well using the latest unstables (10.9.0, build version `2024041505`) after upgrading my instance from 10.8.13 (and previous versions).
> 
> The issue was that the `Path` entries in the `TypedBaseItems` table of the `library.db` for each library did not match the actual filesystem paths.
> 
> My data is in `/srv/jellyfin`, with a symlink from `/var/lib/jellyfin` to `/srv/jellyfin`, which seemed to be enough in 10.8.z. But this resulted in TypedBaseItems entries that looked like this:
> 
> ```
> sqlite> SELECT Path FROM TypedBaseItems WHERE Path LIKE '%/jellyfin/root/default/%';
> /var/lib/jellyfin/root/default/Collections
> /var/lib/jellyfin/root/default/Movies
> /var/lib/jellyfin/root/default/Music
> /var/lib/jellyfin/root/default/Shows
> /var/lib/jellyfin/root/default/YouTube
> ```
> 
> I suspect due to having the data directory path configured wrong when initially setting up the libraries.
> 
> Manually updating these DB entries to the correct paths corrected the issue (i.e. `UPDATE TypedBaseItems SET Path='/srv/jellyfin/root/default/Collections' WHERE Path='/var/lib/jellyfin/root/default/Collections;` et al.). I no longer get the "null name" error and library settings save correctly.

This is super duper weird. I just looked at the code and the returned library info is indeed derived from the stored library path, and will be incomplete if the path does not match. The problem is, such matching mechanism is very, very old and is actually emby code. I have no idea why this ever worked in 10.8.

**Comment by joshuaboniface** [comment]
Agreed there, not sure how it worked fine before. I think we can leave the fix pending for 10.9.1 and see if we really need it based on real-world reports after 10.9.0 drops.

**Comment by unlimitedbacon** [comment]
Just want to add that I am seeing the same issue after upgrading directly from 10.8.13 to 10.9.3

```
Uncaught Error: null name
    value https://xxx/web/node_modules.jellyfin-apiclient.bundle.js?54c888a45bbc6109b58a:2
    p https://xxx/web/98217.72dc08e4589a52ca512a.chunk.js:1
node_modules.jellyfin-apiclient.bundle.js:2:34730
```

**Comment by brianjmurrell** [comment]
And I am seeing it after upgrading from 10.8.13 to 10.9.4.
